contracts/v3/FiatTokenV3.sol

```diff
@@ -0,0 +1,300 @@
+/**
+ * SPDX-License-Identifier: MIT
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+pragma solidity 0.6.12;
+
+import { FiatTokenV2_1 } from "../v2/FiatTokenV2_1.sol";
+import { EIP712 } from "../util/EIP712.sol";
+
+// solhint-disable func-name-mixedcase
+
+/**
+ * @title FiatTokenV3
+ * @dev ERC20 Token backed by fiat reserves
+ */
+contract FiatTokenV3 is FiatTokenV2_1 {
+
+    /**
+     * @notice Initialize v3
+     */
+    function initializeV3(address[] calldata accountsToBlacklist) external {
+        require(_initializedVersion == 2, "v3 initialized out of order");
+
+        // re-add previously blacklisted accounts to the blacklist
+        // by setting the high bit of their balance to 1
+        for (uint256 i = 0; i < accountsToBlacklist.length; i++) {
+            _blacklist(accountsToBlacklist[i]);
+        }
+
+        // additionally blacklist the contract address itself
+        _blacklist(address(this));
+
+        DOMAIN_SEPARATOR = EIP712.makeDomainSeparator(name, "3");
```

Comment:
It would, which I don't think is necessary.

Now that I'm thinking about it, I'm not sure this even needs to be considered a major version bump 😅 . While it made sense I think to call the previous proposal v3, since this is effectively keeping all functionality the same and not changing the interface, I think it might be worth calling it v2.2.

Reply:
Went ahead and made this a minor version bump to 2.2, and kept `DOMAIN_SEPARATOR` unchanged.

https://github.com/centrehq/centre-tokens/pull/357/commits/49a3737121e65a4e75851f25f151eb4812553c25